icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/NumberFormatTest.java

```diff
@@ -7120,4 +7120,63 @@ public TestData(String unitIdentifier, Integer inputValue, UnitWidth width, ULoc
         }
 
     }
+
+    @Test
+    public void TestPortionFormat() {
+        MeasureUnit unit = MeasureUnit.forIdentifier("portion-per-1e9");
+        LocalizedNumberFormatter formatter = NumberFormatter.withLocale(ULocale.ENGLISH).unit(unit)
+                .unitWidth(UnitWidth.FULL_NAME);
+        String formatted1 = formatter.format(1).toString();
+        assertEquals("1 portion per 1e9", "1 part per billion", formatted1);
+
+        String formatted2 = formatter.format(2).toString();
+        assertEquals("1000 portion per 1e9", "2 parts per billion", formatted2);
+
+        String formatted3 = formatter.format(1000000).toString();
+        assertEquals("1000000 portion per 1e9", "1,000,000 parts per billion", formatted3);
+
+        String formatted4 = formatter.format(1000000000).toString();
+        assertEquals("1000000000 portion per 1e9", "1,000,000,000 parts per billion", formatted4);
+    }
+
+    @Test
+    public void TestArbitraryPortionFormat() {
+        class TestData {
+            String unitIdentifier;
+            Integer inputValue;
+
+            TestData(String unitIdentifier, Integer inputValue) {
+                this.unitIdentifier = unitIdentifier;
+                this.inputValue = inputValue;
+            }
+        }
+
+        TestData[] testData = {
+                new TestData("portion-per-1e1", 1),
+                new TestData("portion-per-1e2", 1),
+                new TestData("portion-per-1e3", 1),
+                new TestData("portion-per-1e4", 1),
+                new TestData("portion-per-1e5", 1),
+                new TestData("portion-per-1e6", 1),
+                new TestData("portion-per-1e7", 1),
+                new TestData("portion-per-1e8", 1),
+        };
+
+        for (TestData testCase : testData) {
+            try {
+                MeasureUnit unit = MeasureUnit.forIdentifier(testCase.unitIdentifier);
+                LocalizedNumberFormatter formatter = NumberFormatter.withLocale(ULocale.ENGLISH).unit(unit)
+                        .unitWidth(UnitWidth.FULL_NAME);
+                formatter.format(testCase.inputValue);
+                // NOTE: All these test cases should pass, but because CLDR data is missing for
+                // `portion`, an error is thrown.
+                // See: CLDR-18274
+                fail("Expected IllegalArgumentException for unit: " + testCase.unitIdentifier
```

Comment:
This is the thing you're waiting on Mihai for, right?  Shouldn't you be using `logKnownIssue()` for this?  And don't you already have what you need from Mihai?

Also, I assume when the localized names for "portion" are there, you'll update this to check the actual returned value, right?

Reply:
From Mihai, I have only `portion-per-1e9` data, not `portion` , therefore, I can format `portion-per-1e9` but not `portion-per-XYZ` , see CLDR-18274.

Therefore, I am adding these test cases that expected a failure.

by the way, what is `logKnownIssue` ?

Reply:
>From Mihai, I have only portion-per-1e9 data, not portion , therefore, I can format portion-per-1e9 but not portion-per-XYZ , see [CLDR-18274](https://unicode-org.atlassian.net/browse/CLDR-18274).

Oh, I see.  Okay, then.

>by the way, what is logKnownIssue ?

There's a function called `logKnownIssue()` that you can call to indicate that a particular case is known to fail and that we have a ticket outstanding for fixing it.  I think that's what you want to be using here.  I don't remember the details off the top of my head, but if you search the code base for it, you'll find it being used a lot.

Reply:
okay, I will merge this PR and have a following PR with `logKnownIssue` function